pkg/agent/util/dump.go

```diff
@@ -0,0 +1,90 @@
+package util
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/afero"
+	"k8s.io/utils/exec"
+
+	"github.com/vmware-tanzu/antrea/pkg/ovs/ofctl"
+	"github.com/vmware-tanzu/antrea/pkg/ovs/vsctl"
+)
+
+type Dumper interface {
+	DumpFlows(basedir string) error
+	DumpIPTables(basedir string) error
+	DumpIPToolInfo(basedir string) error
+}
+
+type ofctlClientFactory func(br string) ofctl.OfctlClient
+
+func NewDumper(fs afero.Fs, executor exec.Interface, factory ofctlClientFactory) Dumper {
+	return &dumper{
+		factory:  factory,
+		fs:       fs,
+		executor: executor,
+	}
+}
+
+type dumper struct {
+	factory  ofctlClientFactory
+	fs       afero.Fs
+	executor exec.Interface
+}
+
+func (d *dumper) DumpFlows(basedir string) error {
+	brs, err := vsctl.NewClient(d.executor).ListBridges()
+	if err != nil {
+		return fmt.Errorf("error when collecting ovs bridge info: %w", err)
+	}
+	err = d.fs.Mkdir(filepath.Join(basedir, "flows"), os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("error when creating flows output dir: %w", err)
+	}
+	for _, br := range brs {
+		flows, err := d.factory(br).DumpFlows()
```

Comment:
Did not get why add this factory for ofctl, but not for vsctl you added?

Reply:
Because the fs and executor of vsctl could be replaced.
I have updated the code and merged vsctl to the interface.